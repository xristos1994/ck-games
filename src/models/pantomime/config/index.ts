import { GameStates } from "./GameStates";
import { Movies } from "./Movies";
import { AvailableTimes } from "./AvailableTimes";
import { AvailableScoreTargets } from "./AvailableScoreTargets";

export { GameStates, Movies, AvailableTimes, AvailableScoreTargets };
