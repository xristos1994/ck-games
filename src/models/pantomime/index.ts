import { pantomimeEpic } from './epics';
import { pantomimeReducer } from './reducer';

export { pantomimeEpic, pantomimeReducer };
