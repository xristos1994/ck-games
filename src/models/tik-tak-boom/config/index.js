import { GameStates } from "./GameStates";
import { Modes } from "./Modes";

export { GameStates, Modes };
