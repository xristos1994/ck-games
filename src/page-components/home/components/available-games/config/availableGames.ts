export const availableGames: { name: string; url: string }[] = [
  { name: "Tik-Tak-Boom", url: "/tik-tak-boom" },
  { name: "Pantomime", url: "/pantomime" },
];
