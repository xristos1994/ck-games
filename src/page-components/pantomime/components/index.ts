import { SEO } from "./SEO";
import { PlayersSetup } from "./players-setup";
import { ScoreSetup } from "./score-setup";
import { StartRound } from "./start-round";
import { RoundInProgress } from "./round-in-progress";
import { EndRound } from "./end-round";
import { EndGame } from "./end-game";
import { ScoreBoard } from "./score-board";

export {
  SEO,
  PlayersSetup,
  ScoreSetup,
  StartRound,
  RoundInProgress,
  EndRound,
  EndGame,
  ScoreBoard,
};
