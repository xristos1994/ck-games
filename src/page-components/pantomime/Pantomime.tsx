import React, { FC, ReactElement, useEffect } from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  SEO,
  TeamsSetup,
  ScoreSetup,
  AvailableTimeSetup,
  StartRound,
  RoundInProgress,
  EndRound,
  EndGame,
} from "./components";
import { gameState } from "@models/pantomime/props";
import { initializeGame } from "@models/pantomime/actions";
import { GameStates } from "@models/pantomime/config";
import { IState } from "@models/interfaces";
import styles from "./styles.module.css";

interface IProps {
  gameState: GameStates;
  initializeGame: () => void;
}

const _Pantomime: FC<IProps> = ({
  gameState,
  initializeGame,
}): ReactElement => {
  useEffect(() => {
    initializeGame();
  }, [initializeGame]);

  const playerSetup = gameState === GameStates.setTeams && <TeamsSetup />;

  const scoreSetup = gameState === GameStates.setScoreTarget && <ScoreSetup />;

  const availableTimeSetup = gameState === GameStates.setAvailableTime && (
    <AvailableTimeSetup />
  );

  const startRound = gameState === GameStates.waitForRoundStart && (
    <StartRound />
  );

  const roundInProgress = gameState === GameStates.roundInProgress && (
    <RoundInProgress />
  );

  const endGame = gameState === GameStates.gameEnded && <EndGame />;

  const endRound = gameState === GameStates.roundEnded && <EndRound />;

  return (
    <div className={styles.pageBody}>
      <SEO />
      {playerSetup}
      {scoreSetup}
      {availableTimeSetup}
      {startRound}
      {roundInProgress}
      {endRound}
      {endGame}
    </div>
  );
};

const Pantomime = connect(
  createStructuredSelector<
    IState,
    { gameState: IProps["gameState"] },
    { gameState: IProps["gameState"] }
  >({
    gameState,
  }),
  { initializeGame }
)(_Pantomime);

export { Pantomime };
