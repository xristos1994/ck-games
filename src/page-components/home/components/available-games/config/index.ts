import { availableGames } from "./availableGames";

export { availableGames };
