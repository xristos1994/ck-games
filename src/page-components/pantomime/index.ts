import { Pantomime } from "./Pantomime";

export { Pantomime };
