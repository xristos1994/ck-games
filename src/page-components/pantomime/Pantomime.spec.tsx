import React from 'react';
import renderer from 'react-test-renderer';
import { _Pantomime as Pantomime } from './Pantomime';
import { GameStates } from '@models/pantomime/config';

const initializeGame = () => void 0;

const SEO = () => {
  return (
    <div>
      SEO
    </div>
  );
};

const TeamsSetup = () => {
  return (
    <div>
      TeamsSetup
    </div>
  );
};

const ScoreSetup = () => {
  return (
    <div>
      ScoreSetup
    </div>
  );
};

const AvailableTimeSetup = () => {
  return (
    <div>
      AvailableTimeSetup
    </div>
  );
};

const StartRound = () => {
  return (
    <div>
      StartRound
    </div>
  );
};

const RoundInProgress = () => {
  return (
    <div>
      RoundInProgress
    </div>
  );
};

const EndRound = () => {
  return (
    <div>
      EndRound
    </div>
  );
};

const EndGame = () => {
  return (
    <div>
      ScoreSetup
    </div>
  );
};

jest.mock('./components/', () => ({
  __esModule: true,
  SEO: SEO,
  TeamsSetup: TeamsSetup,
  ScoreSetup: ScoreSetup,
  AvailableTimeSetup: AvailableTimeSetup,
  StartRound: StartRound,
  RoundInProgress: RoundInProgress,
  EndRound: EndRound,
  EndGame: EndGame
}));

describe('page-components/Pantomime', () => {
  it('renders correctly', () => {
    let tree = renderer.create(
      <Pantomime
        initializeGame={initializeGame}
        gameState={GameStates.setTeams}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Pantomime
        initializeGame={initializeGame}
        gameState={GameStates.setScoreTarget}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Pantomime
        initializeGame={initializeGame}
        gameState={GameStates.setAvailableTime}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Pantomime
        initializeGame={initializeGame}
        gameState={GameStates.waitForRoundStart}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Pantomime
        initializeGame={initializeGame}
        gameState={GameStates.roundInProgress}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Pantomime
        initializeGame={initializeGame}
        gameState={GameStates.roundEnded}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <Pantomime
        initializeGame={initializeGame}
        gameState={GameStates.gameEnded}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
