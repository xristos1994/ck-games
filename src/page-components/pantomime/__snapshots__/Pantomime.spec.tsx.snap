// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-components/Pantomime renders correctly 1`] = `
<div
  className="pageBody"
>
  <div>
    SEO
  </div>
  <div>
    TeamsSetup
  </div>
</div>
`;

exports[`page-components/Pantomime renders correctly 2`] = `
<div
  className="pageBody"
>
  <div>
    SEO
  </div>
  <div>
    ScoreSetup
  </div>
</div>
`;

exports[`page-components/Pantomime renders correctly 3`] = `
<div
  className="pageBody"
>
  <div>
    SEO
  </div>
  <div>
    AvailableTimeSetup
  </div>
</div>
`;

exports[`page-components/Pantomime renders correctly 4`] = `
<div
  className="pageBody"
>
  <div>
    SEO
  </div>
  <div>
    StartRound
  </div>
</div>
`;

exports[`page-components/Pantomime renders correctly 5`] = `
<div
  className="pageBody"
>
  <div>
    SEO
  </div>
  <div>
    RoundInProgress
  </div>
</div>
`;

exports[`page-components/Pantomime renders correctly 6`] = `
<div
  className="pageBody"
>
  <div>
    SEO
  </div>
  <div>
    EndRound
  </div>
</div>
`;

exports[`page-components/Pantomime renders correctly 7`] = `
<div
  className="pageBody"
>
  <div>
    SEO
  </div>
  <div>
    ScoreSetup
  </div>
</div>
`;
